connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -391,7 +391,15 @@ public List<Map<String, String>> connectorTaskConfigs(String connName, Connector
             Connector connector = workerConnector.connector();
             try (LoaderSwap loaderSwap = plugins.withClassLoader(workerConnector.loader())) {
                 String taskClassName = connector.taskClass().getName();
-                for (Map<String, String> taskProps : connector.taskConfigs(maxTasks)) {
+                List<Map<String, String>> taskConfigs = connector.taskConfigs(maxTasks);
+                try {
+                    checkTasksMax(connName, taskConfigs.size(), maxTasks, connConfig.enforceTasksMax());
+                } catch (TooManyTasksException e) {
+                    // TODO: This control flow is awkward. Push task config generation into WorkerConnector class?
```

Comment:
Yep, probably in `ConnectUtils` so as not to clutter up the `Worker` class unnecessarily.

Reply:
Do you think that you'll address this in this PR or wait for a follow-up?

Reply:
I think it makes sense as a follow-up, because the awkwardness of the control flow mostly stems from having to trigger a call to `WorkerConnector::onFailure` from a different thread than the one that `run` is being executed on. It'd probably make sense to tackle that and making the herder thread resilient against blocks in `Connector::taskConfigs` in the same PR.

Does that sound alright?